ompi/communicator/comm_init.c

```diff
@@ -388,6 +388,15 @@ static void ompi_comm_construct(ompi_communicator_t* comm)
     comm->c_peruse_handles = NULL;
 #endif
     OBJ_CONSTRUCT(&comm->c_lock, opal_mutex_t);
+
+#if OPAL_ENABLE_FT_MPI
+    comm->any_source_enabled  = true;
```

Comment:
I have  a patch that does that, however it made my life very hard debugging the revamp of the request_state_ok function. I'd rather keep it so that we can gdb the state without it being cryptic.